kitsune/questions/jinja2/questions/question_details.html

```diff
@@ -135,9 +135,15 @@ <h2 class="sumo-callout-heading summary {% if product %}no-product-heading{% end
 
         {% if question.is_spam %}
           <p class="is-spam">
-            {% trans marked_by=question.marked_as_spam_by, marked_time=question.marked_as_spam %}
-              Marked as spam by {{ marked_by }} at {{ marked_time }}
-            {% endtrans %}
+            {% if question.marked_as_spam_by %}
+              {% trans marked_by=question.marked_as_spam_by, marked_time=question.marked_as_spam %}
+                Marked as spam by {{ marked_by }} at {{ marked_time }}
+              {% endtrans %}
+            {% else %}
+              {% trans marked_time=question.marked_as_spam %}
+                Marked as spam at {{ marked_time }}
+              {% endtrans %}
+            {% endif %}
```

Comment:
This could be reduced to:
```jinja2
{% trans marked_by=question.marked_as_spam_by|default(''), marked_time=question.marked_as_spam %}
  Marked as spam{% if marked_by %} by {{ marked_by }}{% endif %} at {{ marked_time }}
{% endtrans %}
```

Reply:
I just discovered that control structures like `if` are not allowed in translatable sections, so I'll have to keep it as it is.